src/components/NavBar/NavBar.tsx

```diff
@@ -1,8 +1,115 @@
-import React from 'react';
+import React, { useState } from 'react';
+import {
+  AppBar,
+  IconButton,
+  Toolbar,
+  Button,
+  useMediaQuery,
+} from '@mui/material';
+import {
+  Menu,
+  AccountCircle,
+  Brightness4,
+  Brightness7,
+  ArrowDropDown,
+  Search,
+  Notifications,
+} from '@mui/icons-material';
+import { Link } from 'react-router-dom';
+import { useTheme } from '@mui/material/styles';
+
+import makeStyles from './styles';
 
 function NavBar() {
+  const [profileOpen, setProfileOpen] = useState<boolean>(true);
+  const isMobile = useMediaQuery('(max-width:600px)');
+  const [mobileOpen, setMobileOpen] = useState<boolean>(false);
+
+  const classes = makeStyles();
+  const theme = useTheme();
+  const isAuthenticated = true;
+
+  // TODO: Write function to switch to dark and light modes
+  const switchDarkMode = () => {};
+
   return (
-    <div>NavBar</div>
+    <AppBar position="relative" style={{ background: 'transparent', boxShadow: 'none', color: 'black', width: '100%' }}>
+      <Toolbar className={classes.toolbar}>
+        {isMobile ? (
+          <>
+            <IconButton
+              color="inherit"
+              edge="start"
+              style={{ outline: 'none' }}
+              onClick={() => setMobileOpen((prevMobileOpen) => !prevMobileOpen)}
+              className={classes.menuButton}
+            >
+              <Menu />
+            </IconButton>
+
+            {isAuthenticated && (
+            <div className={classes.searchInput}>
+              <Search /> &nbsp; Search Github...
+            </div>
+            )}
+          </>
+        ) : (
+          <div className={classes.searchGroup}>
+            {isAuthenticated && (
+            <>
+              <IconButton
+                color="inherit"
+                sx={{ ml: 1, mr: 2 }}
+                onClick={switchDarkMode}
+              >
+                {theme.palette.mode === 'dark' ? (
+                  <Brightness7 />
+                ) : (
+                  <Brightness4 />
+                )}
+              </IconButton>
+
+              <div className={classes.searchInput}>
+                <Search /> &nbsp; Search Github...
+              </div>
+            </>
+            )}
+          </div>
+        )}
+
+        <div>
+          {!isAuthenticated ? (
+            <Button color="inherit" onClick={() => {}}>
```

Comment:
`onClick` does nothing.
Rather stub it and add a TODO comment like you have here:
```ts
// TODO: Write function to switch to dark and light modes
const switchDarkMode = () => {};
```